pkg/remote/kyma_synchronization_context.go

```diff
@@ -128,11 +129,33 @@ func (c *KymaSynchronizationContext) ensureRemoteNamespaceExists(ctx context.Con
 	return nil
 }
 
-func (c *KymaSynchronizationContext) CreateOrUpdateCRD(ctx context.Context, plural string) error {
+func UpdateKymaAnnotations(
+	kyma *v1beta2.Kyma,
+	kcpCRD *v1extensions.CustomResourceDefinition,
+	skrCRD *v1extensions.CustomResourceDefinition) {
+	if kyma.Annotations == nil {
+		kyma.Annotations = make(map[string]string)
+	}
+
+	kcpAnnotation := v1beta2.KcpKymaCRDGenerationAnnotation
+	skrAnnotation := v1beta2.SkrKymaCRDGenerationAnnotation
+
+	if kcpCRD.Spec.Names.Kind == string(v1beta2.ModuleTemplateKind) {
+		kcpAnnotation = v1beta2.KcpModuleTemplateCRDGenerationAnnotation
+		skrAnnotation = v1beta2.SkrModuleTemplateCRDGenerationAnnotation
+	}
+
+	kyma.Annotations[kcpAnnotation] = strconv.FormatInt(kcpCRD.Generation, 10)
+	kyma.Annotations[skrAnnotation] = strconv.FormatInt(skrCRD.Generation, 10)
+}
+
+func CreateOrUpdateCRD(
```

Comment:
since it's a shared function, you need to consider put it in to a better place, in this cse, `crd_upgrade.go` is a good candidate.